src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,495 @@
+
+    Delegate::ProvideOfferRequestArgs args;
+    args.id                 = sessionId;
+    args.streamUsage        = req.streamUsage;
+    args.sdp                = req.sdp;
+    args.videoStreamId      = videoStreamID;
+    args.audioStreamId      = audioStreamID;
+    args.iceServers         = req.ICEServers;
+    args.iceTransportPolicy = req.ICETransportPolicy;
+    args.metadataOptions    = req.metadataOptions;
+    args.peerNodeId         = GetNodeIdFromCtx(ctx.mCommandHandler);
+    args.peerFabricIndex    = ctx.mCommandHandler.GetAccessingFabricIndex();
+
+    CHIP_ERROR err = mDelegate->HandleProvideOffer(args, outSession);
+
+    // Convert the CHIP_ERROR to an Interaction Model status code
+    Protocols::InteractionModel::Status status = StatusIB(err).mStatus;
+    if (status != Protocols::InteractionModel::Status::Success)
+    {
+        // If the delegate could not allocate or encountered a problem,
+        // respond with the appropriate status. For example, if out of resources,
+        // the delegate might return an error that maps to ResourceExhausted or Failure.
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    // Store the WebRTCSessionStruct in the CurrentSessions
+    AddOrUpdateSession(outSession);
+
+    // Send the response
+    Commands::ProvideOfferResponse::Type resp;
+    resp.webRTCSessionID = outSession.id;
+    resp.audioStreamID.SetValue(outSession.videoStreamID);
+    resp.audioStreamID.SetValue(outSession.audioStreamID);
+
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, resp);
+}
+
+void WebRTCTransportProviderServer::HandleProvideAnswer(HandlerContext & ctx, const Commands::ProvideAnswer::DecodableType & req)
+{
+    // Extract command fields from the request.
+    uint16_t sessionId = req.webRTCSessionID;
+    auto sdpAnswer     = req.sdp;
+
+    NodeId peerNodeId           = GetNodeIdFromCtx(ctx.mCommandHandler);
+    FabricIndex peerFabricIndex = ctx.mCommandHandler.GetAccessingFabricIndex();
+
+    WebRTCSessionStruct * existingSession = FindSession(sessionId);
+
+    if (!existingSession)
+    {
+        // If WebRTCSessionID is not null and does not match a value in current sessions.
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
+        return;
+    }
+
+    // Also check that the existing session belongs to the same PeerNodeID / Fabric
+    // If it doesn’t match, respond with CONSTRAINT_ERROR
+    if (peerNodeId != existingSession->peerNodeID || peerFabricIndex != existingSession->peerFabricIndex)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
+        return;
+    }
+
+    Status status  = Protocols::InteractionModel::Status::Success;
+    CHIP_ERROR err = mDelegate->HandleProvideAnswer(sessionId, sdpAnswer);
+    if (err != CHIP_NO_ERROR)
+    {
+        status = StatusIB(err).mStatus;
```

Comment:
`CHIP_NO_ERROR` seems to map to succes: https://github.com/project-chip/connectedhomeip/blob/master/src/protocols/interaction_model/StatusCode.cpp#L52

So could we maybe always do:

```cpp
ctx.mCommandHandler.AddStatus(
    ctx.mRequestPath, 
    ClusterStatusCode(mDelegate->HandleProvideAnswer(sessionId, sdpAnswer))
)
```

Reply:
It seems better to have ClusterStatusCode than converting through StatusIB (which takes the status code as well) and then move back to Status.

Reply:
This comment applies to the other handlers too.